docs/source/explanations/storage-database.md

```diff
@@ -0,0 +1,143 @@
+in particular, which will have one writer and potentially many concurrent readers.
+But we also anticipate use cases for streaming _processed_ data, which may have
+multiple concurrent writers as well. Directly backing this with files would be
+intractable: we need a database.
+
+## Which database?
+
+The "catalog" and authentication database are straightforward traditional OLTP
+(OnLine Transaction Processing) databases. Most queries are looking for a
+needle in a hackstack: querying for a single row or a page of ~100 rows.
+
+The use case for Tiled's "storage" database may include some OLAP (OnLine
+Analytical Processing) workloads, fetching or performing reduction queries
+on a large fraction of the rows in a table. The data tends to be numerical,
+and expressing correct numerical precision (e.g. uint8 versus "integer")
+may be important for fidelity, speed, and storage efficiency.
+
+The space of "OLAP" database is a large and growing space, currently receiving
+a lot of attention. Well-known projects include [DuckDB](https://duckdb.org/)
+and [ClickHouse](https://clickhouse.com/); there are many, many others in the
+category.
+
+To start, Tiled supports these three backends:
+
+- PostgreSQL (recommended for production)
+- DuckDB (default for single-process dev)
+- SQLite
+
+Of these, only DuckDB is OLAP, and notably it can only be connected to a single
+process at a time, so it is not compatible with production, horizontally-scaled
+Tiled deployments.
+
+We recommend PostgreSQL for production because the _stewardship_ of the data is
+the paramount concern, and PostgreSQL is extremely well trusted not to lose or
+corrupt data. We received a strong recommendation from professional SQL developers
+familiar with our use case to start with PostgreSQL. We are likely to stay there
+for the next 1–2 years at least.
+
+We are watching other projects, including the DuckDB integration with PostgreSQL
+which would leverage DuckDB's richer numerical type system within PostgreSQL.
+
+## What schema?
+
+When Bluesky emits an Event Descriptor, it defines a schema for the Events (rows)
+to follow. These schemas are often repeated, but can be unique, defined by
+some arbitrary collection of measured signals.
+
+Tiled receives this as an Arrow schema (names and data types) followed by
+batches of rows. It supplements these rows with a `dataset_id` and a
+`partition_id` and inserts them into a table. By default, it uses a table
+with a name derived from the schema, `table_{arrow_schema_hash}`, and thus
+`dataset_id` is necessary to identify the rows comprising logically related
+data, e.g. coming from the same Bluesky run. The client has the option to
+specify a meaningful, human-readable name if preferred. If a matching table
+does not already exist, Tiled creates it. (This too, could be customized.
+Perhaps some instruments would want to only use tables that are pre-declared
+and purposefully named.)
```

Comment:
This paragraph would greatly benefit from a figure: A schematic of a table(s) with data from a toy experiment that uses multiple schema.